cli/node/mod.rs

```diff
@@ -1039,19 +1038,29 @@ fn is_relative_specifier(specifier: &str) -> bool {
 }
 
 fn file_extension_probe(
-  mut p: PathBuf,
+  p: PathBuf,
   referrer: &Path,
 ) -> Result<PathBuf, AnyError> {
-  if p.exists() && !p.is_dir() {
-    Ok(p.clean())
-  } else {
-    p.set_extension("js");
-    if p.exists() && !p.is_dir() {
-      Ok(p)
+  let p = p.clean();
+  if p.exists() {
+    let mut p_js = p.clone();
+    p_js.set_extension("js");
+    if p_js.exists() {
```

Comment:
The `!p_js.is_dir()` check is important. You can have some packages called "something.js" and they will have that as a directory.

Reply:
I think it's not important in this case, we are trying to resolve a file within a package directory not a package itself, have you seen a package that has "foo.js" as a directory inside it?

Reply:
Yup. I have. That's why I added that check 😄

Reply:
I forget the package name that had that. Maybe it would be good to add a comment saying why the `!.is_dir()` check is necessary.

Reply:
Okay, so I think in this branch the check is not necessary - it's the crux of this fix. I hit that `!p.is_dir()` check trying to translate `chai`. 

in `chai@x.x.x/lib` there is `chai.js` and `chai/`; doing `!p.is_dir()` makes it impossible to ever resolve a file in that directory.

The check in second branch seems to be fine